scripts/functions/logging

```diff
@@ -87,69 +104,121 @@ rvm_pretty_print()
       esac
       return 0
       ;;
+
     (2|force)
       return 0
       ;;
   esac
 }
 
-case "${TERM:-dumb}" in
-  (dumb|unknown)
-    rvm_error_clr=""
-    rvm_warn_clr=""
-    rvm_debug_clr=""
-    rvm_notify_clr=""
-    rvm_reset_clr=""
-    ;;
-  (*)
-    __rvm_set_color rvm_error_clr  "${rvm_error_color:-red}"
-    __rvm_set_color rvm_warn_clr   "${rvm_warn_color:-yellow}"
-    __rvm_set_color rvm_debug_clr  "${rvm_debug_color:-magenta}"
-    __rvm_set_color rvm_notify_clr "${rvm_notify_color:-green}"
-    __rvm_set_color rvm_reset_clr  "${rvm_reset_color:-reset}"
-    ;;
-esac
+__rvm_set_colors()
+{
+  case "${TERM:-dumb}" in
+    (dumb|unknown)
+      rvm_error_clr=""
+      rvm_warn_clr=""
+      rvm_debug_clr=""
+      rvm_notify_clr=""
+      rvm_code_clr=""
+      rvm_comment_clr=""
+      rvm_reset_clr=""
+      ;;
+    (*)
+      __rvm_set_color rvm_error_clr   "${rvm_error_color:-red}"
+      __rvm_set_color rvm_warn_clr    "${rvm_warn_color:-yellow}"
+      __rvm_set_color rvm_debug_clr   "${rvm_debug_color:-magenta}"
+      __rvm_set_color rvm_notify_clr  "${rvm_notify_color:-green}"
+      __rvm_set_color rvm_code_clr    "${rvm_code_color:-blue}"
+      __rvm_set_color rvm_comment_clr "${rvm_comment_color:-iblack}"
+      __rvm_set_color rvm_reset_clr   "${rvm_reset_color:-reset}"
+      ;;
+  esac
+}
+
+__rvm_replace_colors()
+{
+  \typeset ___text
+
+  ___text="${1//<error>/$rvm_error_clr}"
+  ___text="${___text//<warn>/$rvm_warn_clr}"
+  ___text="${___text//<debug>/$rvm_debug_clr}"
+  ___text="${___text//<notify>/$rvm_notify_clr}"
+  ___text="${___text//<code>/$rvm_code_clr}"
+  ___text="${___text//<comment>/$rvm_comment_clr}"
+  ___text="${___text//<log>/$rvm_reset_clr}"
+
+  ___text="${___text//<\/error>/$rvm_reset_clr}"
+  ___text="${___text//<\/warn>/$rvm_reset_clr}"
+  ___text="${___text//<\/debug>/$rvm_reset_clr}"
+  ___text="${___text//<\/notify>/$rvm_reset_clr}"
+  ___text="${___text//<\/code>/$rvm_reset_clr}"
+  ___text="${___text//<\/comment>/$rvm_reset_clr}"
+  ___text="${___text//<\/log>/$rvm_reset_clr}"
+
+  printf "%b" "${___text}$rvm_reset_clr"
+}
 
-exec 6>&2
 rvm_printf_to_stderr()
 {
   printf "$@" >&6
 }
+
 rvm_error()
 {
   if rvm_pretty_print stderr
-  then rvm_printf_to_stderr "%b" "${rvm_error_clr:-}$*${rvm_reset_clr:-}\n"
-  else rvm_printf_to_stderr "%b" "$*\n"
+  then printf "%b" "$(__rvm_replace_colors "<error>$*</error>\n")" >&6
```

Comment:
double `printf "%b"` it should be enough to:
```bash
then __rvm_replace_colors "<error>$*</error>\n" >&6
```
and this happens in few places bellow too

Reply:
Yes, but then would ` >&6` work well?